File: .cpp
#include <iostream>
using namespace std;

int main() {
    unsigned long long N;
     cin >> N;

    bool stan_turn = true;  // Вказує, чи хід Слави (true) або Олі (false)

    while (N > 1) {
      if (stan_turn) {
        N = (N + 8) / 9;  // Максимальне значення, на яке можна поділити
        } else {
        N = (N + 1) / 2;  // Мінімальне значення, на яке можна поділити
        }
         stan_turn = !stan_turn;  // Зміна черги гравця
    }

        if (!stan_turn) {
          cout << "Stan wins." << endl;
          } else {
          cout << "Ollie wins." << endl;
          }

           return 0;
}

File: .js
const readline = require('readline');

const rl = readline.createInterface({
 input: process.stdin,
 output: process.stdout
 });

function whoWins(N) {
 let stanTurn = true; // Вказує, чи хід Слави (true) або Олі (false)

    while (N >= 2) {
      if (stanTurn) {
        N = Math.ceil(N / 9); // Максимальне значення, на яке можна поділити
        } else {
        N = Math.ceil(N / 2); // Мінімальне значення, на яке можна поділити
        }
        stanTurn = !stanTurn; // Змінюємо чергу гравця
        }

      if (!stanTurn) {
        console.log("Stan wins.");
        } else {
        console.log("Ollie wins.");
    }
}

let lineNumber = 1;
rl.on('line', (input) => {
 if (lineNumber === 1) {
   const N = parseInt(input.trim(), 10);
    whoWins(N);
     rl.close();
    }
      lineNumber++;
});
